### PR TITLE
DEVPROD-3471: update deps

### PR DESCRIPTION
**Pull request by jayakasadev**
* upgrade to go 1.21
* update deps
* remove deprecated code

### CODE REVIEW

**Comment by julianedwards** [comment]
evergreen retry

File: pprof.go [hunk 1]
```diff
@@ -88,7 +85,7 @@ func profile(w http.ResponseWriter, r *http.Request) {
 		fmt.Fprintf(w, "Could not enable CPU profiling: %s\n", err)
 		return
 	}
-	sleep(w, time.Duration(sec)*time.Second)
+	sleep(r, time.Duration(sec)*time.Second)
```
Comment (ybrill) [inline pprof.go:88]: I wonder: could these use the [time.Sleep](https://pkg.go.dev/time#Sleep) function?

Comment (jayakasadev) [reply pprof.go:88]:
```
func sleep(r *http.Request, d time.Duration) {
	ctx := r.Context()
	select {
	case <-time.After(d):
	case <-ctx.Done():
	}
}
```
sleep tries to terminate early if the request context closes

**Review by ybrill — APPROVED** [review]

File: pprof.go (hunk 1 above)
Comment (ybrill) [reply pprof.go:88]: ah, that makes sense.
Thanks for explaining!

**Pull request merged**